modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php

```diff
@@ -109,4 +110,63 @@ public function testDemoContent() {
     $assert->linkNotExists('Ludwig van Beethoven');
   }
 
+  /**
+   * Ensures sidebar blocks only appear when they have content.
+   */
+  public function testSidebarLayoutClass() {
+    $this->config('block.block.az_barrio_sidebar_menu')->set('status', FALSE)->save();
+
+    $fixture_path = __DIR__ . '/../../fixtures/config';
+    $storage = new FileStorage($fixture_path);
+    $config_name = 'block.block.az_demo_test_sidebar_menu';
+    $config_data = $storage->read($config_name);
+    $this->assertNotEmpty($config_data, 'Sidebar menu fixture config found.');
+    $this->config($config_name)->setData($config_data)->save();
+    $this->container->get('entity_type.manager')->getStorage('block')->resetCache();
+
+    // Front page should not display the sidebar menu.
+    $this->drupalGet(Url::fromRoute('<front>'));
+    $assert = $this->assertSession();
+    $assert->statusCodeEquals(200);
+    $body = $assert->elementExists('css', 'body');
+    $classes = $body->getAttribute('class');
+    $this->assertStringNotContainsString('layout-one-sidebar', $classes, 'Since the block is empty, the `layout-one-sidebar` class should NOT be present on the front page.');
+    $this->assertStringNotContainsString('layout-sidebar-first', $classes, 'Since the block is empty, the `layout-sidebar-first` class should NOT be present on the front page.');
+    $this->assertStringContainsString('layout-no-sidebars', $classes, 'Since the block is empty, the `layout-no-sidebars` class SHOULD be present on the front page.');
+    $assert->elementNotExists('css', '#block-az-demo-test-sidebar-menu');
+
```

Comment:
```suggestion
    $assert->elementNotExists('css', '#block-az-demo-test-sidebar-menu');
    $classes = $body->getAttribute('class');
    $this->assertStringNotContainsString('layout-one-sidebar', $classes, 'Since the block is empty, the `layout-one-sidebar` class should NOT be present on the front page.');
    $this->assertStringNotContainsString('layout-sidebar-first', $classes, 'Since the block is empty, the `layout-sidebar-first` class should NOT be present on the front page.');
    $this->assertStringContainsString('layout-no-sidebars', $classes, 'Since the block is empty, the `layout-no-sidebars` class SHOULD be present on the front page.');
```